app/models/piece.rb

```diff
@@ -42,6 +55,12 @@ def obstructed?(destination_x, destination_y)
 
   private
 
+  def remove_piece(destination_x, destination_y)
+    destination_piece = Piece.find_by column_coordinate: destination_x, row_coordinate: destination_y
+    destination_piece[:is_on_board?] = false
+    destination_piece
```

Comment:
^ Yes. The current code isn't saving the record after making the change. Can do what @RaykPlatzek said and it should work.